src/main/java/seedu/address/model/lesson/MakeUpLesson.java

```diff
@@ -44,5 +44,22 @@ public boolean isRecurring() {
         return false;
     }
 
+    /**
+     * Returns true if this {@code MakeUpLesson} clashes with the given {@code Lesson}.
+     *
+     * @param otherLesson The other lesson to be compared with.
+     * @return True if and only if lessons clash.
+     */
+    @Override
+    public boolean isClashing(Lesson otherLesson) {
+        if (otherLesson.isRecurring()) {
+            return getLocalDate().compareTo(otherLesson.getLocalDate()) >= 0 // same date or after
+                    && getDayOfWeek().equals(otherLesson.getDayOfWeek()) // same day
+                    && getTimeRange().isClashing(otherLesson.getTimeRange());
```

Comment:
I think can split into 3 boolean functions? `hasDateClash`, `hasDayClash` and `hasTimeClash` so it is easier to read

Reply:
perhaps should put these functions in the parent `Lesson` class? since all of the other child classes require it as well like `RecurringLesson`

Reply:
imo, `hasDateClash(getLocalDate(), otherLesson.getLocalDate())` isn't that much more readable haha